packages/@jsii/go-runtime/jsii-runtime-go/runtime.go

```diff
@@ -286,8 +287,18 @@ func castAndSetToPtr(ptr interface{}, data interface{}, implMap implementationMa
 		client := getClient()
 		client.objects[ptrVal.Interface()] = ref.InstanceID
 	} else {
-		val := reflect.ValueOf(data)
-		ptrVal.Set(val)
+		// Attempt to unmarshal (again) with the concrete return type. This gives
+		// custom unmarshal logic (e.g. for $jsii.enums) an opportunity to be discovered
+		// based on the actual return type.
+		bytes, err := json.Marshal(data)
+		if err != nil {
+			panic(err)
+		}
+
+		err = json.Unmarshal(bytes, ptr)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
I'm not sure why enum types wouldn't be caught by first pass deserialization unless we weren't declaring the type signature correctly in certain places. Are they being seen as target type `interface{}` on the first pass or "string" (in the case of string enums?). If the latter then this makes sense to me.

Reply:
The `Result` field in the "Invoke" response structure is `interface{}` (which makes sense), and that's the type we unmarshal from the wire.

Reply:
That looks a little late to me and I kinda wish we'd have a way that did not involve re-marshaling, then re-unmarshaling here. This is also not gonna work (guessing) for enum entries returned by an `any` method/property?

Reply:
Indeed, this won't work if the return type if `any`. Any suggestion on how to tackle?